chain/chain/src/store.rs

```diff
@@ -2295,7 +2295,7 @@ impl<'a> ChainStoreUpdate<'a> {
                 if outcomes_with_id.is_empty() {
                     self.gc_col(DBCol::TransactionResult, outcome_id.as_bytes());
                 } else {
-                    store_update.set_ser(
```

Comment:
You will need to write another merge operator here for removing an outcome_id from the column, which is what this operation is doing here. Otherwise, we are still doing a read-modify-write here, which has the same problem with the other place. Unfortunately, gc is also still in the main ClientActor thread, so it will be problematic if the writes here spends a lot of time.

Reply:
Perhaps we need to change the column so that it stores `Vec<TransactionResultOp>` where:

```rust
pub enum TransactionResultOp {
    Add(ExecutionOutcomeWithIdAndProof),
    Del(CryptoHash)
}
```

and then during merging interpret the deletion and add compaction filter which gets rid of the value if the vector is empty.  This would be a bit more convoluted to read but would allow us to remove the value without reading it first.

It might be useful to quickly sketch out a draft solution with a new column using `(outcome_id || black_hash)` and then prefix scan for reads.  (It also occurred to me that ExecutionOutcomeWithIdAndProof contains both the outcome_id and block_hash which can be decoded from the key so perhaps further optimisation could be to have some kind of serialisation where the two hashes are not repeated in the value.  Then again, perhaps that’s too much work at least for the time being).